src/main/scala/rocket/Frontend.scala

```diff
@@ -355,7 +353,7 @@ trait HasICacheFrontend extends CanHavePTW { this: BaseTile =>
   // don't actually use the device that is instantiated in the frontend.
   private val deviceOpt = if (tileParams.icache.get.itimAddr.isDefined) Some(frontend.icache.device) else None
 
-  val iCacheLogicalTreeNode = new ICacheLogicalTreeNode(frontend.icache, deviceOpt, tileParams.icache.get)
+  val iCacheLogicalTreeNode = new ICacheLogicalTreeNode(() => frontend.icache.module.data_arrays.map(_._2), deviceOpt, tileParams.icache.get)
```

Comment:
Not really. The tool simply rewrites the call to `ICacheLogicalTreeNode.apply (...)` to match the args.
Should you add the `"-Yno-adapted-args"` to scalac, this code won't compile at all.

This change looks fine to me, since `ICacheLogicalTreeNode.apply` takes a function as the first argument and you are passing a value, which is implicitly adopted.

```sca.a
class ICacheLogicalTreeNode(memories: () => Seq[OMSRAM], deviceOpt: Option[SimpleDevice], icacheParams: ICacheParams) extends LogicalTreeNode(() => deviceOpt) {}
```